controller/Lists.php

```diff
@@ -71,31 +69,31 @@ class Lists extends tao_actions_CommonModule
     private $isListsDependencyEnabled;
 
     /**
-     * Show the list of users
+     * Show the list
      * @return void
      */
     public function index()
     {
         tao_helpers_Scriptloader::addCssFile(Template::css('lists.css', 'tao'));
         $this->defaultData();
 
-        $myAdderFormContainer = new tao_actions_form_List();
-        $myForm = $myAdderFormContainer->getForm();
-
-        if ($myForm->isSubmited()) {
-            if ($myForm->isValid()) {
-                $values = $myForm->getValues();
-                $newList = $this->getListService()->createList($values['label']);
-                $i = 0;
-                while ($i < $values['size']) {
-                    $this->getListService()->createListElement($newList, __('element') . ' ' . ($i + 1));
-                    $i++;
-                }
-            }
-        } else {
-            $myForm->getElement('label')->setValue(__('List') . ' ' . (count($this->getListService()->getLists()) + 1));
+        $this->setData('lists', $this->getListData());
+        $this->setView('Lists/index.tpl');
+    }
+
+    /**
+     * Action called when the user creates a new list.
+     * @return void
+     */
+    public function newLocalList() {
```

Comment:
As we discussed @hectoras, idea: 

1 - Return a 201 status code with json content only (Do not return HTML here).
2 - FE call 'index' endpoint again after receive the success.

Reply:
Done :point_right: 5fd76e3c898de97f3755c8cd32c2fbe4088e0eb3

Reply:
1c9e4803fd1894b266243f32101ff4381bf3af2c makes changes in this controller to return a JSON payload (with a 201 HTTP status code) for AJAX POST calls on this endpoint. The frontend will use the response payload redraw the page.

The payload returned from the backend will be in the form:

```
{
   "success":true,
   "data":{
      "newList":{
         "name":"List 4",
         "uri":"https:\/\/tao.host\/ontologies\/tao.rdf#i61a4cfb5a1ec28325f877e99b8d704"
      }
   }
}
```

Reply:
Marking this thread as solved, let's discuss additional details for the current implementation (single method, returning JSON for POST) on the latest thread.